tests/configurable_cluster.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Cluster object used to control a cluster that can be configured with
+a yaml file.
+
+Test writers should use this module for all of their cluster related needs.
+"""
+
+import fnmatch
+import os
+import paramiko
+from subprocess import check_call
+import tempfile
+import yaml
+
+from prestoadmin import main_dir
+from tests.product import user_password
+
+CONFIG_FILE_GLOB = r'*.yaml'
+DEFAULT_MOUNT_POINT = '/mnt/presto-admin'
+DIST_DIR = os.path.join(main_dir, 'tmp/installer')
+
+
+class ConfigurableCluster(object):
+    """Start/stop/control/query a cluster defined by a configuration file.
+
+    This class allows you to run the presto-admin product tests on a real
+    cluster.
+
+    """
+    def __init__(self, config_filename):
+        # see PyDoc for all_internal_hosts() for an explanation on the
+        # difference between an internal and regular host
+        with open(os.path.join(main_dir, config_filename)) as config_file:
+            config = yaml.load(config_file)
+
+        self.master = config['master']
+        self.slaves = config['slaves']
+        self.internal_master = config['internal_master']
+        self.internal_slaves = config['internal_slaves']
+        if self.internal_master != 'master' or \
```

Comment:
If it has to have a particular value, it's not really configurable.  I think set it in the code, and put a note in the pydoc that that's a requirement (just like sudo access).  You should still check that self.master and self.slaves are appropriately sized.